### PR TITLE
Add pre-commit hook

### PR DESCRIPTION
**Pull request by kidrahahjo**
## Description

Enabling the support of [pre-commit](https://pre-commit.com/) for pre-commit hook.
Developers will now have to perform these steps in order to enable the pre-commit hook.
```
pip install -r requirements-precommit.txt
pre-commit install
```
This will add the hook to `git`.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

**Comment by mikemhenry** [comment]
See https://github.com/glotzerlab/signac-docs/pull/92 for how to setup pre-commit

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 click>=7.0
 ruamel.yaml>=0.15.89
 flake8>=3.7.7
```
Comment (jennyfothergill) [inline requirements-dev.txt:3]: Why do we pin two different flake8 versions in requirements-precommit and requirements-dev?

Comment (kidrahahjo) [reply requirements-dev.txt:3]: This is not meant to be in this file.
I'll remove this.

**Comment by kidrahahjo** [comment]
@jennyfothergill I agree but the developers use `flake8` only before committing. Hence, I think we should keep all the `pre-checks` dependencies in a separate requirements file.
I proposed this for signac [here](https://github.com/glotzerlab/signac/pull/358#issuecomment-662323328) and it was implemented in this manner by @bdice for signac.

**Comment by bdice** [comment]
@jennyfothergill Good question! The reason we split it is so that our CI doesn't have to install all the dependencies just to run the pre-checks. It speeds things up.

**Pull request merged**